Suggestion of a new nest parameter in healpy.smoothing()

MaudeCharmetant opened the pull request:
I suggest implementing a new "nest" parameter in healpy.sphtfunc.smoothing() so that when the input map is in NESTED ordering, the function takes care of reordering it to RING for the smoothing and back to NESTED before the output. I have tried to follow the way "nest" argument where used in other Healpy functions, but maybe the description in the docstring can be improved and maybe the ordering back to NESTED can be removed and you could just specify that healpy.sphtfunc.smoothing() will return a RING ordered map no matter the input. 

I hope this helps a bit or at least serves as a base to improve the clarity of the smoothing function.

zonca commented on healpy/sphtfunc.py:
```diff
@@ -943,6 +944,11 @@ def smoothing(
     verbose : bool, optional
       If True prints diagnostic information. Default: True
+    nest : bool, optional
+      If True, ordering scheme is NESTED. Default: False (RING)
```
ordering scheme "of the input map is assumed to be NESTED"

zonca commented on healpy/sphtfunc.py:
```diff
@@ -943,6 +944,11 @@ def smoothing(
       If True prints diagnostic information. Default: True
+    nest : bool, optional
+      If True, ordering scheme is NESTED. Default: False (RING)
+      The map ordering is modified by this function, the input map array should
```
`reorder` makes copy of the array, so we are not modifying the input.
however I agree we need to mention that internally we are running `reorder`

zonca commented on healpy/sphtfunc.py:
```diff
@@ -1015,6 +1027,12 @@ def smoothing(
+        ordering = "NESTED"
+        output_map = pixelfunc.reorder(output_map, inp=None, out=None, r2n=True, n2r=False)
+    else:
+        ordering = "RING" 
```
we probably dont need to redefine `ordering` here.

MaudeCharmetant replied:
Yes true, thank you! I have removed the "ordering" and also the else condition which is not essential I suppose.

zonca reviewed (commented):
very good, thanks for your help with this! I added feedback for minor improvements

zonca commented:
about reordering the output, let's keep that, we don't want by default the function to have an output ordering different than the input ordering.

can you also add a line to `CHANGELOG.rst` mentioning this?

MaudeCharmetant commented:
Thank you for all of your comments! I have implemented them, let me know if we need further changes, especially on the phrasing of things. I have updated CHANGELOG.rst but I think I was supposed to update it in my branch (which I did afterward) and not submit a new pull request, sorry.

zonca commented:
thanks for your contribution, much appreciated!

MaudeCharmetant commented:
Thank you for Healpy, this is such a great tool!
